pkg/daemon/ceph/client/config.go

```diff
@@ -133,6 +140,28 @@ func generateConfigFile(context *clusterd.Context, clusterInfo *ClusterInfo, pat
 	return filePath, nil
 }
 
+func mergeDefaultConfigWithRookConfigOverride(context *clusterd.Context, clusterInfo *ClusterInfo, configFile *ini.File) error {
+	cm, err := context.Clientset.CoreV1().ConfigMaps(clusterInfo.Namespace).Get(k8sutil.ConfigOverrideName, metav1.GetOptions{})
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed to read configmap %q", k8sutil.ConfigOverrideName)
+		}
+		return nil
+	}
+
+	config, ok := cm.Data["config"]
+	if !ok {
+		logger.Debugf("No ceph configuration override to merge as %q configmap is empty", k8sutil.ConfigOverrideName)
+		return nil
+	}
+
+	if err := configFile.Append([]byte(config)); err != nil {
```

Comment:
Don't you need to trim the global section? If not, won't you get a duplicate [global] section?.

Reply:
If you mean the trim if it looks like that `"    [global]   \n    bluestore_min_alloc_size_hdd = 4096"` it is not necessary it well ends up with only one global section
If it something else could you please provide an ex. so I can check

Reply:
I mean, the `ceph.conf` from the prepare pod looks something like:

```
[global]
foo = bar
...
...
```

Then if the configmap content is:

```
[global]
bluestore_min_alloc_size_hdd = 4096
...
...
```

Won't we can a file like this?

```
[global]
foo = bar
...

[global]
bluestore_min_alloc_size_hdd = 4096
...
...
```

I'm assuming this because the method is called `Append()` but it does the trimming for is too then it's good.
Have you verified that?

Thanks!

Reply:
Yes I've tested it and the output file looks like (no global section duplicated)
```
[global]
foo = bar
bluestore_min_alloc_size_hdd = 4096
...
```

Reply:
Looks good in my testing as well